bundles/org.openhab.binding.innogysmarthome/src/main/java/org/openhab/binding/innogysmarthome/internal/handler/InnogyBridgeHandler.java

```diff
@@ -335,26 +339,28 @@ private void setPropertyIfPresent(final String key, final @Nullable Object data,
     @Override
     public void dispose() {
         logger.debug("Disposing innogy SmartHome bridge handler '{}'", getThing().getUID().getId());
-
         unregisterDeviceStatusListener(this);
-
-        if (reinitJob != null) {
-            reinitJob.cancel(true);
-            reinitJob = null;
-        }
-
+        cancelReinitJob();
         if (webSocket != null) {
             webSocket.stop();
             webSocket = null;
         }
-
         client = null;
         deviceStructMan = null;
 
         super.dispose();
         logger.debug("innogy SmartHome bridge handler shut down.");
     }
 
+    private synchronized void cancelReinitJob() {
+        ScheduledFuture<?> reinitJob = this.reinitJob;
+
+        if (reinitJob != null) {
+            reinitJob.cancel(true);
+            reinitJob = null;
```

Comment:
Should be `this.reinitJob = null;`.

```suggestion
            this.reinitJob = null;
```